std/algorithm/searching.d

```diff
@@ -1409,59 +1415,65 @@ if (isInputRange!Range && !isInfinite!Range &&
     return extremum!(a => a, selector)(r, seedElement);
 }
 
-@safe pure unittest
+@safe pure @nogc unittest
 {
     // allows a custom map to select the extremum
-    assert([[0, 4], [1, 2]].extremum!"a[0]" == [0, 4]);
-    assert([[0, 4], [1, 2]].extremum!"a[1]" == [1, 2]);
+    import std.array : staticArray;
+    int[2] a = [0, 4];
+    int[2] b = [1, 2];
+    auto c = [a, b].staticArray;
+    assert(c[].extremum!"a[0]" == [0, 4]);
+    assert(c[].extremum!"a[1]" == [1, 2]);
 
     // allows a custom selector for comparison
-    assert([[0, 4], [1, 2]].extremum!("a[0]", "a > b") == [1, 2]);
-    assert([[0, 4], [1, 2]].extremum!("a[1]", "a > b") == [0, 4]);
+    assert(c[].extremum!("a[0]", "a > b") == [1, 2]);
+    assert(c[].extremum!("a[1]", "a > b") == [0, 4]);
 
     // use a custom comparator
     import std.math : cmp;
-    assert([-2., 0, 5].extremum!cmp == 5.0);
-    assert([-2., 0, 2].extremum!`cmp(a, b) < 0` == -2.0);
+    assert([-2., 0, 5].staticArray[].extremum!cmp == 5.0);
+    assert([-2., 0, 2].staticArray[].extremum!`cmp(a, b) < 0` == -2.0);
 
     // combine with map
     import std.range : enumerate;
-    assert([-3., 0, 5].enumerate.extremum!(`a.value`, cmp) == tuple(2, 5.0));
-    assert([-2., 0, 2].enumerate.extremum!(`a.value`, `cmp(a, b) < 0`) == tuple(0, -2.0));
+    assert([-3., 0, 5].staticArray[].enumerate.extremum!(`a.value`, cmp) == tuple(2, 5.0));
+    assert([-2., 0, 2].staticArray[].enumerate.extremum!(`a.value`, `cmp(a, b) < 0`) == tuple(0, -2.0));
 
     // seed with a custom value
     int[] arr;
     assert(arr.extremum(1) == 1);
 }
 
-@safe pure nothrow unittest
+@safe pure nothrow @nogc unittest
 {
     // 2d seeds
-    int[][] arr2d;
-    assert(arr2d.extremum([1]) == [1]);
-
-    // allow seeds of different types (implicit casting)
-    assert(extremum([2, 3, 4], 1.5) == 1.5);
+    import std.array : staticArray;
+    import std.range : only;
+    int[0][0] arr2d;
+    auto d = [2, 3, 4].staticArray;
```

Comment:
I think `.staticArray` should be reserved for rvalues and explicit `int[N]` used for variables (in this case, `int[3]`).